### ISSUE TITLE
Missing translation for : YTD && %1Y

### ISSUE THREAD

**Issue by jamesst20** [amcharts/amcharts4#2607]
Version: "@amcharts/amcharts4": "^4.9.26"

The plugin "am4plugins_rangeSelector.DateAxisRangeSelector" comes with inputs and buttons to change the chart's zoom level.

I applied '@amcharts/amcharts4/lang/fr_FR' and it did translate input's labels, but it did not translate the '1Y' (1 year) button or the 'YTD' (Year to date) button.

Expected:

1Y = 1A
YTD = AAJ

Thank you

**Comment by martynasma** [comment]
Yes, because official language files do not have translations for plugins. You can do it like this:

```
chart.language.locale["%1Y"] = "%1A";
chart.language.locale["YTD"] = "AAJ";
```

**Comment by jamesst20** [comment]
> chart.language.locale["%1Y"] = "%1A";
> chart.language.locale["YTD"] = "AAJ";

Thank you for the quick reply. I have though of that and oddly, it doesn't work.

I also did try doing this

```
selector = new am4plugins_rangeSelector.DateAxisRangeSelector();
...
const fr = /*"am4plugins_rangeSelector.DateAxisRangeSelector"*/
fr["%1Y"] = "%1A";
fr["YTD"] = "AAJ";
chart.language.locale = fr;
selector.language.locale = fr;
```

Calling 
```

**Comment by jamesst20** [comment]
> chart.language.locale["%1Y"] = "%1A";
> chart.language.locale["YTD"] = "AAJ";

Thank you for the quick reply. I have though of that and oddly, it doesn't work.

I also did try doing this

```
selector = new am4plugins_rangeSelector.DateAxisRangeSelector();
...
const fr = /*"am4plugins_rangeSelector.DateAxisRangeSelector"*/
fr["%1Y"] = "%1A";
fr["YTD"] = "AAJ";
chart.language.locale = fr;
selector.language.locale = fr;
```

Calling 
```
selector.language.translateAny('YTD') => AAJ
```

but still displays YTD. Looks like it's loaded when the constructor is instantiated.

**Comment by martynasma** [comment]
Try moving those translations **before** initializing the RangeSelector.

**Comment by jamesst20** [comment]
@martynasma 

Results is the same. The plugin doesn't pick the language of the chart. I have to manually assign it after its initialization.

**Comment by jamesst20** [comment]
> Can you post your whole chart online? I'd like to take a look.

Sure ! It's hard to get a full working example as it is embedded inside a Wordpress.

I extracted the usefull stuff:
https://pastebin.com/E7Ckqk9N

**Comment by martynasma** [comment]
Thanks.

It seems that `RangeSelector` is still before prompt settings:

```
let selector = new am4plugins_rangeSelector.DateAxisRangeSelector();
```

Can you try moving it after?

**Comment by jamesst20** [comment]
@martynasma 

If I do that, the graph is english because it doesn't switch language.

**Comment by jamesst20** [comment]
In the mean time I am going to use it like this while this bug is being resolved:

```
                        let selector = new am4plugins_rangeSelector.DateAxisRangeSelector();
                        if (document.querySelector('html').lang == 'fr-FR') {
                            this.chart.language.locale = am4lang_fr_FR;
                            selector.language.locale = am4lang_fr_FR;
                            selector.periods = selector.periods.map((p) => {
                                if (/^\dY$/i.test(p.name)) return { ...p, name: p.name.replace('Y', 'A') };
                                else if(/^YTD$/i.test(p.name)) return { ...p, name: 'AAJ' };
                                else return p;
                            });
                        } else {
                            this.chart.language.locale = am4lang_en_CA;
                            selector.language.locale = am4lang_en_CA;
                        }
```

**Comment by martynasma** [comment]
I don't think you need to set locale for both chart and selector. Setting it for chart should be enough.

Just make sure it's set before anything else.

**Comment by jamesst20** [comment]
> I don't think you need to set locale for both chart and selector. Setting it for chart should be enough.
> 
> Just make sure it's set before anything else.

Like I said, it doesn't pick the chart's language, it becomes english.

**Comment by martynasma** [comment]
I'm not sure how this could happen.

I'm gonna have to ask you to post your whole chart on CodePen, so that we can test your setup. Not just a fraction of code.

**Comment by martynasma** [comment]
We finally got to the bottom of this issue. Reopening. Will provide more details when new version is released next week.

**Comment by martynasma** [comment]
Fixed in [4.9.35](https://github.com/amcharts/amcharts4/releases/tag/4.9.35).

## [4.9.35] - 2020-08-12

### Added
- New adapter on `XYSeries`: `groupValue`. It's now possible to apply custom aggregate value calculation functions for grouped data items. [More info](https://www.amcharts.com/docs/v4/tutorials/using-custom-functions-for-data-item-grouping/).
- New adapter on `Cursor`: `cursorPoint`. Use it to custom-position chart cursor using your own logic.

### Changed
- Accessibility: Now if ENTER is pressed when object with an enabled `"hit"` event is focused, that event will be triggered automatically.
- Clicking anywhere on the plot area with `XYCursor` will now trigger blur on any focused element on the page (consistent to clicking anywhere else on the document).

### Fixed
- RangeSelector plugin was not properly inheriting chart's locale.
- `"x"` (timestamp) in `inputDateFormat` was not working properly.
- If all values of `SankeyDiagram` were `0` (zero), nodes were not being arranged and their rectangles were not being displayed.
- First `CategoryAxis` label was not visible if `minGridDistance` was set to `0` (zero).
- Old series bullets of an `XYSeries` remained visible when data was updated and data grouping was enabled on related `DateAxis`.
- `Utils.random()` function was not returning value in proper from/to range.
- Stacking now works only between series of the same type. Previously a `LineSeries` between two stacked `ColumnSeries` could mess up stacking for the latter.
- Zooming in on a part of `LineSeries` which had `strokeDashArray` enabled via `propertyFields` was making the whole line dashed.
- On a chart with `XYCursor` hovering on category with a null value for some series, used to display a tooltip for a different category in some cases.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
